### PR TITLE
Add Authorisation to POST and PUT endpoints

### PR DESCRIPTION
**Pull request by justinpjose**
### What

**Describe what you have changed and why.**
Add Zebedee Client
Add Authorisation to POST and PUT endpoints

### How to review

**Describe the steps required to test the changes.**
1. Run the tests in the api package
2. Run the API **locally** (`make debug`) and carry out some curl requests:

_Without Authorisation Header_ [Return Status Code 401]
`curl -i -X PUT http://localhost:22300/recipes/b944be78-f56d-409b-9ebd-ab2b77ffe187 -d '{"alias":"test"}'`

_With Authorisation Header but invalid token_ [Return Status Code 401]
`curl -i -X PUT http://localhost:22300/recipes/b944be78-f56d-409b-9ebd-ab2b77ffe187 -d '{"alias":"test"}' -H "Authorization: wrong"`

_With Authorisation Header and valid token_ [Return Status Code 200]
`curl -i -X PUT http://localhost:22300/recipes/b944be78-f56d-409b-9ebd-ab2b77ffe187 -d '{"alias":"test"}' -H "Authorization: $SERVICE_AUTH_TOKEN"`

### Who can review

**Describe who worked on the changes, so that other people can review.**
Anyone - Justin worked on the changes

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -45,23 +64,25 @@ func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration,
...
+		dataStore:         dataStore,
+		Router:            router,
+		recipePermissions: recipePermissions,
+		permissions:       permissions,
+		EnableAuthImport:  cfg.MongoConfig.EnableAuthImport,
 	}
 
 	api.get("/health", api.HealthCheck)
 	api.get("/recipes", api.RecipeListHandler)
 	api.get("/recipes/{id}", api.RecipeHandler)
```
Comment (morses-code) [inline api/api.go:78]: `curl -L -X GET 'http://localhost:22300/recipes'` returns 401 Unauthorized code, my understanding is that the GET methods shouldn't need auth headers? Maybe someone else could clarify this.

File: api/api.go [hunk 2]
```diff
@@ -1,35 +1,54 @@
...
-func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration, dataStore store.DataStore, hc *healthcheck.HealthCheck, errorChan chan error) {
+func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration, dataStore store.DataStore, hc *healthcheck.HealthCheck, errorChan chan error, recipePermissions AuthHandler, permissions AuthHandler) {
 	router := mux.NewRouter()
-	api := NewRecipeAPI(ctx, cfg, router, dataStore)
+	api := NewRecipeAPI(ctx, cfg, router, dataStore, recipePermissions, permissions)
 
 	healthcheckHandler := newMiddleware(hc.Handler)
 	middleware := alice.New(healthcheckHandler)
 
+	// Add the identity middleware when running in publishing
```
Comment (nshumoogum) [inline api/api.go:49]: This always gets run, whether in publishing or web?

Comment (justinpjose) [reply api/api.go]: This removed in commit "remove identity middleware to avoid authenticating whole API"

File: api/api.go (hunk 1 above)
Comment (justinpjose) [reply api/api.go:74]: Problem resolved in commit "remove identity middleware to avoid authenticating whole API"

**Pull request merged**